python/lsst/pipe/analysis/utils.py

```diff
@@ -1281,20 +1289,18 @@ def makeBadArray(catalog, flagList=[], onlyReadStars=False, patchInnerOnly=True,
         bad |= ~catalog["detect_isPatchInner"]
     if "detect_isTractInner" in schema and tractInnerOnly:
         bad |= ~catalog["detect_isTractInner"]
+
     if "deblend_scarletFlux" in schema:
-        # Exclude parents that are blends of multiple sources
-        bad |= ((catalog["parent"] == 0) & (catalog["deblend_nChild"] > 1))
-        # Exclude isolated sources not passed to deblender
-        if excludeIsolatedNotModel:
+        if "detect_isDeblendedSource" in schema:
```

Comment:
I see that you intentionally left backwards compatibility in this PR. I recommend fixing this going forward, and having a separate branch to work with legacy catalogs. Otherwise please create a ticket to strip out the unneccessary code and leave a TODO comment here with the ticket number.

Reply:
Ok, that's fair.